robotbase/src/main/java/frc/robot/commands/drive/DriveToReef.java

```diff
@@ -0,0 +1,368 @@
+package frc.robot.commands.drive;
+
+import static edu.wpi.first.units.Units.Degrees;
+import static edu.wpi.first.units.Units.Meters;
+import static frc.robot.Constants.DRIVE_TO_POSE.FINAL_APPROACH_DISTANCE;
+import static frc.robot.Constants.DRIVE_TO_POSE.INTERPLOATION_PERCENT;
+import static frc.robot.Constants.DRIVE_TO_POSE.ROTATION_TOLERANCE;
+import static frc.robot.Constants.DRIVE_TO_POSE.X_TOLERANCE;
+import static frc.robot.Constants.DRIVE_TO_POSE.Y_TOLERANCE;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_A;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_B;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_C;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_D;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_E;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_F;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_G;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_H;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_I;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_J;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_K;
+import static frc.robot.Constants.FIELD.REEF.BRANCH_L;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants.DRIVE_TO_POSE.COLLISION_AVOIDANCE;
+import frc.robot.Constants.FIELD.REEF;
+import frc.robot.Robot;
+import frc.robot.controls.controllers.ButtonboardController.ReefSide;
+import frc.robot.controls.controllers.ButtonboardController.ScoringDirection;
+import frc.robot.util.Utility;
+import java.util.function.Function;
+
+public class DriveToReef extends Command {
+
+  private Pose2d m_currPose;
+
+  private Pose2d m_currentTarget;
+  private Pose2d m_lastTarget;
+
+  private Pose2d m_finalGoal;
+
+  private DriveToPose m_currDriveToPose;
+
+  private Twist2d m_collisionAvoidanceTwist = new Twist2d();
+
+  private boolean m_isDriverControlling;
+
+  public DriveToReef() {}
+
+  @Override
+  public void initialize() {
+    m_lastTarget = Robot.swerve.getAllianceRelativePose2d();
+    m_finalGoal = getGoalFromButtonboard();
```

Comment:
We should be able to assume this is not null, or it returns an `Optional<Pose2d>`